New Resource: aws_iot_thing

radeksimko opened the pull request:
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSIotThing_ -timeout 120m
=== RUN   TestAccAWSIotThing_basic
--- PASS: TestAccAWSIotThing_basic (59.38s)
=== RUN   TestAccAWSIotThing_full
--- PASS: TestAccAWSIotThing_full (377.55s)
=== RUN   TestAccAWSIotThing_importBasic
--- PASS: TestAccAWSIotThing_importBasic (33.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	470.151s
```

bflad commented on aws/resource_aws_iot_thing_test.go:
```diff
@@ -0,0 +1,159 @@
+		}
+
+		_, err := conn.DescribeThing(params)
+		if err == nil {
```
Should we flip this around to return `nil` on `iot.ErrCodeResourceNotFoundException` error and return the actual error on others (e.g. permission errors)?

bflad commented on website/docs/r/iot_thing.html.markdown:
```diff
@@ -0,0 +1,36 @@
+    Creates and manages an AWS IoT Thing.
+---
+
+# aws_iot_thing
```
Missing `## Import` documentation

radeksimko commented on aws/resource_aws_iot_thing.go:
```diff
@@ -0,0 +1,158 @@
+		Delete: resourceAwsIotThingDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```
Yeah, that's a piece of code I somehow forgot here as I was refactoring the CRUD.

radeksimko commented:
@bflad PTAL, I believe I addressed all your comments 😉 

```
=== RUN   TestAccAWSIotThing_basic
--- PASS: TestAccAWSIotThing_basic (18.44s)
=== RUN   TestAccAWSIotThing_full
--- PASS: TestAccAWSIotThing_full (351.33s)
=== RUN   TestAccAWSIotThing_importBasic
--- PASS: TestAccAWSIotThing_importBasic (19.77s)
```

bflad commented on aws/resource_aws_iot_thing_test.go:
```diff
@@ -0,0 +1,194 @@
+		}
+
+		_, err := conn.DescribeThing(params)
+		if err != nil {
```
Aw shucks - just missing a `return err` in here, otherwise we print out the wrong message below

bflad commented:
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
